### PR TITLE
Removing (most) Closure from BlockFactory

### LINKED ISSUE #668: BlocklyFactory: Remove dependency on closure.
The main ones to remove are goog.dom.xml and goog.ui.PopupColorPicker.

Once that's done, remove the message in app_controller's init.

### PR DESCRIPTION
**Pull request by AnmAtAnm**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#668 (partially)

### Proposed Changes

Replacing all Closure `goog.` library calls from BlockFactory, with the exception of the ColorPicker and the test for the warning about running without Closure. This includes (almost) all `goog.provide` and `goog.require` statements.

Final status:
```
$ grep -rF "goog." .
./index.html:      Add it only if you need it.  See also goog.require in blockly.js.
./app_controller.js:goog.require('goog.dom.xml');  // Used to detect Closure
./app_controller.js:  if (!window.goog.dom.xml) {
./workspacefactory/wfactory_init.js:goog.require('goog.ui.PopupColorPicker');
./workspacefactory/wfactory_init.js:goog.require('goog.ui.ColorPicker');
./workspacefactory/wfactory_init.js:  var colourPicker = new goog.ui.ColorPicker();
./workspacefactory/wfactory_init.js:  var popupPicker = new goog.ui.PopupColorPicker(null, colourPicker);
./workspacefactory/wfactory_init.js:  goog.events.listen(popupPicker, 'change', function(e) {
```

I'll remove the ColorPicker and warning in a follow-up PR.

### Reason for Changes

Recent changes in Closure have been breaking BlockFactory, which uses Closure libraries but was not compiled by the Closure Compiler. This also simplifies use for people host their own BlockFactory, or running the local file from the repo.

### Test Coverage

Every changed function and class was run at least once (manual testing), verifying I did not break its primary behavior. Most changes are trivial.

Tested on:
 * Desktop Chrome

### CODE REVIEW

File: demos/blockfactory/block_exporter_tools.js [hunk 1]
```diff
@@ -167,45 +157,6 @@ BlockExporterTools.prototype.addBlockDefinitions = function(blockXmlMap) {
   eval(blockDefs);
 };
 
-/**
- * Pulls information about all blocks in the block library to generate XML
- * for the selector workpace's toolbox.
- * @param {!BlockLibraryStorage} blockLibStorage Block Library Storage object.
- * @return {!Element} XML representation of the toolbox.
- */
-BlockExporterTools.prototype.generateToolboxFromLibrary
```
Comment (picklesrus) [inline demos/blockfactory/block_exporter_tools.js]: was this unused?

File: demos/blockfactory/block_library_storage.js [hunk 2]
```diff
@@ -60,17 +57,15 @@ BlockLibraryStorage = function(blockLibraryName, opt_blocks) {
  * Reads the named block library from local storage and saves it in this.blocks.
  */
 BlockLibraryStorage.prototype.loadFromLocalStorage = function() {
-  // goog.global is synonymous to window, and allows for flexibility
-  // between browsers.
-  var object = goog.global.localStorage[this.name];
+  var object = localStorage[this.name];
```
Comment (picklesrus) [inline demos/blockfactory/block_library_storage.js]: none of the browsers require this to be window.?

File: demos/blockfactory/block_library_view.js [hunk 3]
```diff
@@ -59,10 +53,10 @@ var BlockLibraryView = function() {
...
 BlockLibraryView.prototype.addOption = function(blockType, selected) {
   // Create option.
-  var option = goog.dom.createDom('a', {
-    'id': 'dropdown_' + blockType,
-    'class': 'blockLibOpt'
-  }, blockType);
+  var option = document.createElement('a');
+  option.id ='dropdown_' + blockType;
+  option.classList.add('blockLibOpt');
+  option.innerHTML = blockType;
```
Comment (picklesrus) [inline demos/blockfactory/block_library_view.js]: we try and avoid innerHtml. What is this? can it be option.textNode = blockType (or whatever it actually is)

File: demos/blockfactory/block_option.js [hunk 4]
```diff
@@ -70,48 +67,42 @@ var BlockOption = function(blockSelector, blockType, previewBlockXml) {
...
+  labelContainer.classList.add('blockOption_label');
   checkLabelContainer.appendChild(labelContainer);
 
   // Create and append text node for the label.
-  var labelText = goog.dom.createDom('p', {
-    'id': this.blockType + '_text'
-  }, this.blockType);
+  var labelText = document.createElement('p');
+  labelText.id = this.blockType + '_text';
+  labelText.innerHTML = this.blockType;
```
Comment (picklesrus) [inline demos/blockfactory/block_option.js]: same innerHtml comment here.

File: demos/blockfactory/block_definition_extractor.js [hunk 5]
```diff
@@ -164,16 +158,17 @@ BlockDefinitionExtractor.factoryBase_ = function(block, name) {
...
-  // TODO: Deal with colors that don't map to standard hues. (Needs improved
-  //     block definitions.)
-  var colour_hue = Math.floor(
-      goog.color.hexToHsv(block.colour_)[0]);  // Off by one... sometimes
-  var colourBlock = BlockDefinitionExtractor.colourBlockFromHue_(colour_hue);
-  var colourInputValue =
-      BlockDefinitionExtractor.newDomElement_('value', {name: 'COLOUR'});
-  colourInputValue.append(colourBlock);
-  factoryBaseEl.append(colourInputValue);
+  var colour_hue = block.getHue();  // May be null if not set via hue.
```
[the remark this answers is not in the document]
Comment (AnmAtAnm) [reply demos/blockfactory/block_definition_extractor.js]: #1247 was solved by #1293

File: demos/blockfactory/block_exporter_tools.js (hunk 1 above)
Comment (AnmAtAnm) [reply demos/blockfactory/block_exporter_tools.js]: Yes.

File: demos/blockfactory/block_library_storage.js (hunk 2 above)
Comment (AnmAtAnm) [reply demos/blockfactory/block_library_storage.js]: window is the global scope in a browser context: https://stackoverflow.com/a/12660102/152543

File: demos/blockfactory/color_picker.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Color picker used to select the color of a toolbox category.
```
Comment (picklesrus) [inline demos/blockfactory/color_picker.js]: did you mean to add this file in this commit?

Comment (AnmAtAnm) [reply demos/blockfactory/color_picker.js]: Ummm...    what file? :eyes:




(removed)

**Comment by picklesrus** [comment]
LGTM

**Pull request merged**